lucetc/tests/wasm.rs

```diff
@@ -508,14 +528,76 @@ mod module_data {
         );
         */
     }
+
+    #[test]
+    fn names_local() {
+        let m = load_wat_module(
+            "names_local",
+            LoadOptions {
+                write_debug_names: true,
+            },
+        );
+        let b = super::test_bindings();
+        let h = HeapSettings::default();
+        let c = Compiler::new(
+            &m,
+            OptLevel::default(),
+            CpuFeatures::default(),
+            &b,
+            h,
+            false,
+            &None,
+        )
+        .expect("compile local_names");
+        let mdata = c.module_data().unwrap();
+
+        assert_eq!(mdata.import_functions().len(), 0);
+        assert_eq!(mdata.export_functions().len(), 0);
+        assert_eq!(mdata.function_info().len(), 3);
+        assert_eq!(
+            mdata.function_info().get(0).unwrap().name,
+            Some("func_name_0")
+        )
+    }
+
+    #[test]
+    fn names_export() {
```

Comment:
oh! also, happy to add a test here to make sure that the export name is used as a fallback if no names section exists, wasn't sure if that was overkill or not.